### DISCUSSION TITLE
Similarity scores calculation using info from contig txt files

### DISCUSSION THREAD

**Discussion by maxnest** [antismash/antismash#608] [Q&A]
Dear colleagues, 
Thank you for your excellent and very useful approach!
Currently, I test the latest version (7.0.0) of antiSMASH and have one question about of similarity scores displayed in the HTML output under most similar known clusters. If I understand it right, score represent the percent of significant BLAST hits from the number of genes in the subject cluster? Can I calculate these scores using only the information in the txt-files stored in the knownclusterblast directory? Also interested to know if there are any to extract all this information from JSON?
Thank you!

**Comment by SJShaw** [comment]
The similarity percentage is the percentage of genes in that particular reference that have a match (via BLAST) to any gene in the region antiSMASH has detected that have at least 40% coverage and 45% identity (for knownclusterblast specifically, these values are different for other parts of clusterblast).

The `knownclusterblast.txt` file does not contain enough information to regenerate that similarity percentage, since it does not record the number of genes in each reference. It is a legacy output from the days of antiSMASH 4 and earlier, it will likely be removed at some point now that the JSON exists.

The JSON output contains all the information needed to rebuild the similarity for all the hits shown in the HTML. 
`records[].modules["antismash.modules.clusterblast"].knowncluster.results[]` will give you the results for each region with hits in each record. Within that entry, the relevant properties are:
- `region_number`: the region within the record (e.g. record 4, region 2 will be listed here as `2`)
- `ranking`: the first element of this array is the best hit, each element is a tuple of two objects
The first part of that tuple is the summary of the MIBiG entry, the second is a breakdown of the hits themselves.
To recalculate the similarity, divide `hits` from the breakdown (the second element) by the total number of genes in the reference (`proteins` in the first element).
